src/content/protocols/pendle.md

```diff
@@ -0,0 +1,415 @@
+defillama_slug: ["pendle"]
+chain: "Ethereum"
+stage: "O"
+reasons: ["Unverified Contracts"]
+risks: ["L", "H", "M", "H", "H"]
+author: ["CookingCryptos, PaulDeFi"]
+submission_date: "2025-02-19"
+publish_date: "1970-01-01"
+update_date: "1970-01-01"
+---
+
+# Summary
+
+Pendle is a decentralized, permissionless yield-trading protocol that enables users to manage and optimize their yield exposure. By tokenizing yield-bearing assets into standardized yield tokens (SY), Pendle allows these assets to be split into principal tokens (PT) and yield tokens (YT). This separation facilitates various strategies, including earning fixed yields, speculating on future yield changes, and providing liquidity to earn additional rewards. Pendle's Automated Market Maker (AMM) is specifically designed to handle the trading of these time-decaying assets, ensuring efficient and flexible yield management within the decentralized finance ecosystem.
+
+# Ratings
+
+## Chain
+
+Pendle is deployed on various chains. This review is based on the Ethereum mainnet deployment of the protocol.
+
+> Chain score: Low
+
+## Upgradeability
+
+The Pendle V2 protocol is fully upgradeable through multiple proxy patterns including `ERC1967/UUPS` proxies for most core contracts and `TransparentUpgradeableProxy` contracts managed by a central `ProxyAdmin`. This comprehensive upgradeability capability can potentially result in loss of user funds, theft of unclaimed rewards, and significant changes to the protocol's economic parameters and performance.
+
+The permission to upgrade the protocol follows a hierarchical structure. The `Governance`, a 2/4 multisig that serves as the `DEFAULT_ADMIN_ROLE` of the `PendleGovernanceProxy`, has ultimate authority over the protocol's contracts. It directly controls the `ProxyAdmin` (which manages all `TransparentUpgradeableProxy` contracts) and also has direct control over various other contracts including voting mechanisms, market factories, and cross-chain messaging. The `DevMultisig`, a 2/3 multisig, controls fee distribution systems and reward distribution mechanisms like `PendleFeeDistributorV2` and `PendleMultiTokenMerkleDistributor`.
+
+### Core Protocol Components
+
+The `BaseSplitCodeFactoryContract` creates large contracts by splitting their bytecode into two separate fragments that function together as a single logical entity. The resulting fragment contracts remain unverified on Etherscan, creating significant transparency issues as their source code is inaccessible to the public. This contract is controlled by the EOA "Pendle: Deployer 1" and is used to deploy contracts like `PendleMarketFactoryV3`.
+
+The `ProxyAdmin` contract controls all `TransparentUpgradeableProxy` contracts in the protocol through functions including `upgrade`, `upgradeAndCall`, and `changeProxyAdmin`. These functions enable complete replacement of contract logic while preserving state data, with no timelock protection.
```

Comment:
This is too much technical detail, and readers wouldn't know which contracts (AMM? YT? PT?) are of type `TransparentUpgradeableProxy`

Reply:
and saying it can be upgraded is enough, listing exact functions like  `upgrade`, `upgradeAndCall`, and `changeProxyAdmin` is too much detail